plugins/outputs/http/http.go

```diff
@@ -0,0 +1,182 @@
+  keepalive = 0
+  ## Configure HTTP expect continue timeout in seconds. Default : 0
+  expect_continue_timeout = 3
+  ## Configure idle connection timeout in seconds. Default : 0
+  idle_conn_timeout = 3
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+const (
+	POST         = "POST"
+	CONTENT_TYPE = "Content-Type"
+
+	DEFAULT_RETRY                   = 3
+	DEFAULT_CONTENT_TYPE            = "application/json"
+	DEFAULT_TLS_HANDSHAKE_TIMEOUT   = 10
+	DEFAULT_RESPONSE_HEADER_TIMEOUT = 3
+	DEFAULT_DIAL_TIME_OUT           = 3
+	DEFAULT_KEEP_ALIVE              = 3
+	DEFAULT_EXPECT_CONTINUE_TIMEOUT = 3
+	DEFAULT_IDLE_CONN_TIMEOUT       = 3
+)
+
+type Http struct {
+	// http required option
+	URL string `toml:"url"`
+
+	// http default value가 있는 option
+	ContentType           string `toml:"content_type"`
+	Retry                 int    `toml:"retry"`
+	TLSHandshakeTimeout   int    `toml:"tls_handshake_timeout"`
+	ResponseHeaderTimeout int    `toml:"response_header_timeout"`
+	DialTimeOut           int    `toml:"dial_timeout"`
+	KeepAlive             int    `toml:"keepalive"`
+	ExpectContinueTimeout int    `toml:"expect_continue_timeout"`
+	IdleConnTimeout       int    `toml:"idle_conn_timeout"`
+
+	client     http.Client
+	serializer serializers.Serializer
+}
+
+func (h *Http) SetSerializer(serializer serializers.Serializer) {
+	h.serializer = serializer
+}
+
+// Connect to the Output
+func (h Http) Connect() error {
+	h.client = http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   time.Duration(h.DialTimeOut) * time.Second,
+				KeepAlive: time.Duration(h.KeepAlive) * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   time.Duration(h.TLSHandshakeTimeout) * time.Second,
+			ResponseHeaderTimeout: time.Duration(h.ResponseHeaderTimeout) * time.Second,
+			ExpectContinueTimeout: time.Duration(h.ExpectContinueTimeout) * time.Second,
+			IdleConnTimeout:       time.Duration(h.IdleConnTimeout) * time.Second,
+		},
+	}
+
+	return nil
+}
+
+// Close is not implemented. Because http.Client not provided connection close policy. Instead, uses the response.Body.Close() pattern.
+func (h Http) Close() error {
+	return nil
+}
+
+// Description A plugin that can transmit metrics over HTTP
+func (h Http) Description() string {
+	return "A plugin that can transmit metrics over HTTP"
+}
+
+// SampleConfig provides sample example for developer
+func (h Http) SampleConfig() string {
+	return sampleConfig
+}
+
+// Writes metrics over HTTP POST
+// If the request is failed, try again until retry limit.
+func (h Http) Write(metrics []telegraf.Metric) error {
+	if h.URL == "" {
+		return errors.New("Http Output URL Option is empty! It is necessary.")
+	}
+
+	for _, metric := range metrics {
+		buf, err := h.serializer.Serialize(metric)
```

Comment:
instead of writing each metric as an individual buffer, you should add the option for the http output to specify a byte output max buffer size.

that way, the plugin will batch metrics together up to a certain buffer size.

The reason for that is because even a moderate telegraf load would cause a massive number of http requests going over the wire.